go/logic/applier.go

```diff
@@ -502,6 +516,11 @@ func (this *Applier) readMigrationMaxValues(tx *gosql.Tx, uniqueKey *sql.UniqueK
 		return err
 	}
 
+	err = this.setOptimizer(tx)
```

Comment:
@cyrinux looks like `setOptimizer` is ran in both `readMigrationMinValues` and `readMigrationMaxValues`, which means twice in the same SQL transaction

These 2 x funcs are exclusively called [by `ReadMigrationRangeValues`](https://github.com/github/gh-ost/blob/71df3e8d8affca587ee5237391c89533420a5ba3/go/logic/applier.go#L541-L558) which passes both funcs the same transaction

Can we run `setOptimizer` only once in `ReadMigrationRangeValues` [after this line here](https://github.com/github/gh-ost/blob/71df3e8d8affca587ee5237391c89533420a5ba3/go/logic/applier.go#L567)?

Reply:
Sure, thanks for the review !